webapp/lib/app/charts.rb

```diff
@@ -0,0 +1,112 @@
+require 'jbuilder'
+require 'elasticsearch'
+require 'app/charts_helpers'
+
+module Microsite
+
+  class Charts
+
+    include ChartsHelper
+
+    def self.fetch(attribute)
+      body = Jbuilder.encode do |json|
+        json.size 0
+        json.aggs do
+          json.timestamps do
+            json.date_histogram do
+              json.field "@timestamp"
+              json.interval find_interval
+            end
+            json.aggs do
+              json.series do
+                json.terms do
+                  json.field 'label.raw'
+                  json.size 10
+                end
+                json.aggs do
+                  json.stats do
+                     json.stats do
+                       json.field attribute
+                     end
+                  end
+                end
+              end
+            end
+          end
+        end
+      end
+      to_chartjs(client.search(:index => 'logstash-*', :body => body))
```

Comment:
maybe fetch should just do client.search, and to_charjs could be called in the sinatra action?